python/lib/client/dmod/client/client_config.py

```diff
@@ -0,0 +1,119 @@
+import yaml
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Optional
+
+
+class ClientConfig(ABC):
+    _CONFIG_KEY_DATA_SERVICE = 'data-service'
+    _CONFIG_KEY_HOSTNAME = 'hostname'
+    _CONFIG_KEY_PORT = 'port'
+    _CONFIG_KEY_REQUEST_SERVICE = 'request-service'
+    _CONFIG_KEY_SSL_DIR = 'ssl-dir'
+
+    def __init__(self, *args, **kwargs):
+        super(ClientConfig, self).__init__(*args, **kwargs)
+
+    @property
+    @abstractmethod
+    def config_file(self) -> Path:
+        pass
+
+    @property
+    @abstractmethod
+    def dataservice_endpoint_uri(self) -> Optional[str]:
+        pass
+
+    @property
+    @abstractmethod
+    def dataservice_ssl_dir(self) -> Optional[Path]:
+        pass
+
+    @property
+    @abstractmethod
+    def requests_endpoint_uri(self) -> str:
+        pass
+
+    @property
+    @abstractmethod
+    def requests_ssl_dir(self) -> Path:
+        pass
+
+    @abstractmethod
+    def print_config(self):
+        pass
+
+
+class YamlClientConfig(ClientConfig):
+
+    @classmethod
+    def generate_endpoint_uri(cls, hostname: str, port: int):
+        return 'wss://{}:{}'.format(hostname, port)
+
+    @classmethod
+    def get_service_ssl_dir(cls, backing_config: dict, service_key: str):
+        dir_path = Path(backing_config[service_key][cls._CONFIG_KEY_SSL_DIR])
+        if dir_path.is_dir():
+            return dir_path
+        else:
+            raise RuntimeError("Non-existing {} SSL directory configured ({})".format(service_key, dir_path))
+
+    def __init__(self, client_config_file: Path, *args, **kwargs):
+        super(ClientConfig, self).__init__(*args, **kwargs)
+        self._config_file = client_config_file
+        with self._config_file.open() as file:
+            self._backing_config = yaml.safe_load(file)
```

Comment:
it is probably worth a comment here the the class's properties depend on the loading/state of `_backing_config` so they must not be accessed until after the config is loaded

Reply:
I added a few docstrings for the type and the function.